binding/form_mapping.go

```diff
@@ -38,8 +38,6 @@ type setter interface {
 
 type formSource map[string][]string
 
-var _ setter = formSource(nil)
```

Comment:
Please, return back this line. 
And it will be perfect, in my mind, if you add `var _ setter = headerSource(nil)` after [this line](https://github.com/guonaihong/gin/blob/1956-feature-support-bind-http-header/binding/header.go#L28). Thanks.

Reply:
I don't quite understand the meaning of this line of code.
Var _ setter = formSource(nil)
Can you tell me why?
Doesn't it work?

Reply:
check `formSource` implement `setter` interface.

Reply:
It is just checking in compile time, what this type is implemented that interface.

This line is optional, but it is for a good self-documented code. It simply shows why this new type was added. This is a good practice when you add a new type only for a specific interface.

Reply:
### demo
```go
import (
        "reflect"
)

type setOptions struct {
        isDefaultExists bool
        defaultValue    string
}

type setter interface {
        TrySet(value reflect.Value, field reflect.StructField, key string, opt setOptions) (isSetted bool, err error)
}

func mappingByPtr(ptr interface{}, setter setter, tag string) error {
        //_, err := mapping(reflect.ValueOf(ptr), emptyField, setter, tag)
        return nil
}

type formSource map[string][]string

func (f formSource) TrySet(value reflect.Value, field reflect.StructField, key string, opt setOptions) (isSetted bool, err error) {
        return true, nil
}

type headerSource map[string][]string

func main() {

        mappingByPtr(reflect.ValueOf(nil), formSource(map[string][]string{}), "test")
        mappingByPtr(reflect.ValueOf(nil), headerSource(map[string][]string{}), "test")

}
```
### output
./interface.go:32:49: cannot use headerSource(map[string][]string literal) (type headerSource) as type setter in argument to mappingByPtr:
	headerSource does not implement setter (missing TrySet method)

### idea
The compiler has checked when the parameter is assigned. It doesn't seem to be necessary code.
I will check the code tomorrow morning.